### ISSUE TITLE
New format computed property accessor not recognised

### ISSUE THREAD

**Issue by pindab0ter** [barryvdh/laravel-ide-helper#1315]
### Versions:
- ide-helper Version: 2.12.2
- Laravel Version: 9.0.0
- PHP Version: 8.1.2

### Description:

PR #1289 added support for the new attribute mutator/accessor format, but calculated properties/attributes aren't recognised yet.

### Steps To Reproduce:

This example calculated property results in `* @property-read string $full_name`:

```php
public function getFullNameAttribute(): string
{
    return "$this->first_name $this->last_name";
}
```

The following example is not recognised and does not add the same `@property`.

```php
public function fullName(): Attribute
{
    return Attribute::get(fn() => "$this->first_name $this->last_name");
}
```

It should also result in (only) `$fullName` instead of `$full_name`.

The latter should also be `protected` instead of `public` of course, but that's a separate issue (#1293).

**Comment by sawirricardo** [comment]
are there any progress/PR on this?

**Comment by john-inkesta** [comment]
If you make the function public, and provide a return type like so: `fn(): string`, it works.

**Comment by pindab0ter** [comment]
That's true, but then it's no longer an [accessor/mutator](https://laravel.com/docs/10.x/eloquent-mutators#accessors-and-mutators), which is the topic of this issue.

**Comment by john-inkesta** [comment]
What I mean is, if you do this:
```php
public function fullName(): Attribute
{
    return Attribute::get(fn(): string => "$this->first_name $this->last_name");
}
```

It will generate a `full_name` property for you.

**Comment by sawirricardo** [comment]
does it matter though using `public/protected`?

**Comment by john-inkesta** [comment]
> does it matter though using `public/protected`?

The Laravel docs makes them `protected`. But ide-helper only picks them up when they are `public`.

**Comment by pindab0ter** [comment]
> What I mean is, if you do this:
> 
> ```php
> 
> public function fullName(): Attribute
> 
> {
> 
>     return Attribute::get(fn(): string => "$this->first_name $this->last_name");
> 
> }
> 
> ```
> 
> It will generate a `full_name` property for you.
> 
> 
> 
> That's all you want, isn't it?
> 
> 

True. That's what I do nowadays. That said, Laravel supports accessors without typehinting whereas IDE Helper doesn't. 

Hence my PR #1411.

**Comment by TzviPM** [comment]
Any idea when this will be released on packagist? I'd prefer not to directly require `dev-master`, but this does work great for our use case. @barryvdh

**Comment by pindab0ter** [comment]
I have given up on waiting for this and have stopped using this package in favour of the Laravel IDEA plugin.

**Comment by barryvdh** [comment]
A new version was released an hour ago; https://github.com/barryvdh/laravel-ide-helper/releases/tag/v2.14.0
